### PR TITLE
Non exclusive select

### PR DESCRIPTION
**Pull request by nikhilgupta1211**

### CODE REVIEW

File: app/assets/javascripts/event_emails.js.coffee [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
-    $('#event_email_to').val(users)
-    if users.length == 0
-      $('#event_email_to').attr('placeholder', 'No recipients present for ' + state + ' state')
-
-fill_email_to state for state in ['All', 'Accepted', 'Incomplete', 'Submitted', 'Approved', 'Cancel']
+# To remove the duplicate values from an array
+array_unique = (array) ->
+  unique_value = []
+  unique_value.push i for i in array when not (i in unique_value)
+  return unique_value
```
Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: There's probably some function or jquery model available for this. Did you check?

File: app/assets/javascripts/event_emails.js.coffee [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+  return array_unique(value)
+
+# Provide every checkbox with a click event
+$("input[type='checkbox']").each ->
+  $(this).click ->
+    $(this).parent().removeClass('selected')
+    value = []
+    value = users(value)
+    if value.length == 0 && $("input[type='checkbox']:checked").length > 0
+       $('#event_email_to').val('No recipients')
```
Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: Wrong indentation;-)

File: app/assets/javascripts/event_emails.js.coffee [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+      value = value.concat $(this).data('users')
+      $(this).parent().addClass('selected')
+  return array_unique(value)
+
+# Provide every checkbox with a click event
+$("input[type='checkbox']").each ->
+  $(this).click ->
+    $(this).parent().removeClass('selected')
+    value = []
+    value = users(value)
```
Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: Seems you can remove the previous line, right?

**Comment by bgeuken** [comment]
The ui looks fine to me. Please check my comments regarding the code

**Comment by nikhilgupta1211** [comment]
image
image

File: app/assets/javascripts/event_emails.js.coffee (hunk 3 above)
Comment (nikhilgupta1211) [inline app/assets/javascripts/event_emails.js.coffee]: yes

File: app/assets/javascripts/event_emails.js.coffee (hunk 2 above)
Comment (nikhilgupta1211) [inline app/assets/javascripts/event_emails.js.coffee]: thanks :)

File: app/assets/javascripts/event_emails.js.coffee (hunk 1 above)
Comment (nikhilgupta1211) [inline app/assets/javascripts/event_emails.js.coffee]: Sorry but I was not able to found a method to remove duplicate items in a string array :(

Comment (bgeuken) [inline app/assets/javascripts/event_emails.js.coffee]: Okay. Seems not as easy in js as I hoped.

**Pull request merged**